Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/model/BooleanSetting.java

```diff
@@ -39,6 +39,84 @@
+          "ButtonToggleClassicButtonX", true),
+  MAIN_BUTTON_TOGGLE_CLASSIC_3(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleClassicButtonY", true),
+  MAIN_BUTTON_TOGGLE_CLASSIC_4(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleClassicButtonPlus", true),
+  MAIN_BUTTON_TOGGLE_CLASSIC_5(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleClassicButtonMinus", true),
+  MAIN_BUTTON_TOGGLE_CLASSIC_6(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleClassicButtonHome", true),
+  MAIN_BUTTON_TOGGLE_CLASSIC_7(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleClassicTriggerL", true),
+  MAIN_BUTTON_TOGGLE_CLASSIC_8(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleClassicTriggerR", true),
+  MAIN_BUTTON_TOGGLE_CLASSIC_9(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleClassicButtonZL", true),
+  MAIN_BUTTON_TOGGLE_CLASSIC_10(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleClassicButtonZR", true),
+  MAIN_BUTTON_TOGGLE_CLASSIC_11(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleClassicDPad", true),
+  MAIN_BUTTON_TOGGLE_CLASSIC_12(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleClassicStickLeft", true),
+  MAIN_BUTTON_TOGGLE_CLASSIC_13(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleClassicStickRight", true),
+  MAIN_BUTTON_TOGGLE_WII_0(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleWiimoteButtonA", true),
+  MAIN_BUTTON_TOGGLE_WII_1(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleWiimoteButtonB", true),
+  MAIN_BUTTON_TOGGLE_WII_2(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleWiimoteButton1", true),
+  MAIN_BUTTON_TOGGLE_WII_3(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleWiimoteButton2", true),
+  MAIN_BUTTON_TOGGLE_WII_4(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleWiimoteButtonPlus", true),
+  MAIN_BUTTON_TOGGLE_WII_5(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleWiimoteButtonMinus", true),
+  MAIN_BUTTON_TOGGLE_WII_6(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleWiimoteButtonHome", true),
+  MAIN_BUTTON_TOGGLE_WII_7(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleWiimoteDPad", true),
+  MAIN_BUTTON_TOGGLE_WII_8(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleNunchukC", true),
+  MAIN_BUTTON_TOGGLE_WII_9(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleNunchukZ", true),
+  MAIN_BUTTON_TOGGLE_WII_10(Settings.FILE_DOLPHIN, Settings.SECTION_INI_ANDROID,
+          "ButtonToggleNunchukStick", true),
```

Comment:
What do you think about putting these button-specific settings in a separate section? Mostly just to make things less cluttered for people who want to edit the Android section of an INI file manually.

Reply:
The Java-only implementation of adding new sections is gone so I'm not sure how to properly add "AndroidOverlayButtons" section. I tried adding the section string to `Settings` but I couldn't save values to that section without calling `mSettings.saveSettings`, whereas that's unneeded for the "Android" section.

"Android" section is set in `ButtonManager.cpp`. Putting `ini.GetOrCreateSection("AndroidOverlayButtons")` after that didn't work. Any idea what I'm missing for properly making a section? I have a feeling it's obvious.

Reply:
Ah, right, I didn't think of mentioning that problem. You need to add it to this list: https://github.com/dolphin-emu/dolphin/blob/653f89701db560ffe2c9421c81f3b5347c8dc46f/Source/Core/Core/ConfigLoaders/IsSettingSaveable.cpp#L29

Reply:
I just realized that an extra space at the end of "Android" while searching prevented me from finding this :-(